### PR TITLE
Cpu overprovisioner

### PR DESCRIPTION
**Pull request by sj-williams**
- add overprovisioner deployment for CPU requests
- additional proportional-autoscaler for CPU pod count

### CODE REVIEW

File: templates/cpa-cpu.yaml.tpl [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#      - [ 1024, 7 ]
+#      - [ 2048, 10 ]
+#      - [ 4096, 15 ]
+#    nodesToReplicas:
+#      - [ 1, 1 ]
+#      - [ 2, 2 ]
+  linear:
+    nodesPerReplica: 1
+    min: 1
+    max: 100
```
Comment (jaskaransarkaria) [inline templates/cpa-cpu.yaml.tpl:17]: Is this max too high?

File: overprovision.tf [hunk 2]
```diff
@@ -3,19 +3,33 @@
 ##########
 
 locals {
-  pod_memory = {
+  mem_pod_memory = {
```
Comment (jaskaransarkaria) [inline overprovision.tf:6]: these var names are a bit confusing:

mem_pod_memory
mem_pod_cpu
cpu_pod_memory
cpu_pod_cpu

Comment (sj-williams) [reply overprovision.tf]: More meaningful variable names applied

File: templates/cpa-cpu.yaml.tpl (hunk 1 above)
Comment (sj-williams) [reply templates/cpa-cpu.yaml.tpl:17]: Its a ceiling value, that we may not (or may we?) ever reach.

**Review by jaskaransarkaria — APPROVED** [review]

**Pull request merged**